connection/src/traits.rs

```diff
@@ -105,7 +105,7 @@ pub trait UserTxConnection: Connection {
     /// Propose a transaction over the encrypted channel.
     /// Returns the number of blocks in the ledger at the time the call was
     /// received.
-    fn propose_tx(&mut self, tx: &Tx) -> Result<BlockIndex>;
```

Comment:
Why are we changing this from BlockIndex type to u64? I know that they're basically the same, but shouldn't we just use BlockIndex if that's what this number is corresponding to?

Reply:
the problem is that it isn't a block index, it is a count

i find it very confusing to use `BlockIndex` when the number is actually a count, that causes off-by-one mistakes.

i prefer to use u64 everywhere for counts and BlockIndex for indices to help disambiguate

someday we should have type-safety around these to avoid mistakes, but its a large project

Reply:
there's also `BlockCount` in the fog world... https://github.com/mobilecoinfoundation/mobilecoin/blob/bcb3e26cc85afd3a250f2109e9cd388c0e5c814a/fog/types/src/lib.rs#L41

but yeah, u64 is more correct here than BlockIndex.

Reply:
man I forgot about that

Reply:
Except that one lives in fog, so might not make sense for this crate to depend on it...